### PR TITLE
Added specific EnumPPIO for handling enum parameter types.

### PR DESCRIPTION
**Pull request by jdeolive**
I made this change as part of a pull request pending, but I thought I would break it out separately.

### CODE REVIEW

**Comment by aaime** [comment]
Travis says no

**Comment by jdeolive** [comment]
Updated the pull request to handle case where the enum might be private or for whatever other reason unaccessible, and fall back to what was the previous behaviour (ie. LiteralPPIO).

**Comment by jdeolive** [comment]
Travis is happy again so unless there is any other feedback on this one I'll assume it's ok to merge.

File: src/extension/wps/wps-core/src/test/java/org/geoserver/wps/ppio/EnumPPIOTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package org.geoserver.wps.ppio;
```
Comment (aaime) [inline src/extension/wps/wps-core/src/test/java/org/geoserver/wps/ppio/EnumPPIOTest.java]: Lacks copyright header

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/ProcessParameterIO.java [hunk 2]
```diff
@@ -125,7 +125,7 @@
     public static ProcessParameterIO find(Parameter<?> p, ApplicationContext context, String mime) {
         // enum special treatment
         if (p.type.isEnum()) {
-            return new LiteralPPIO(p.type);
+            return new EnumPPIO(p.type);
```
Comment (aaime) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/ProcessParameterIO.java]: Copyright header not updated

Comment (aaime) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/ProcessParameterIO.java]: Also, findAll is still using LiteralPPIO

**Comment by aaime** [comment]
Minor copyright header issues, and one use case for enums in ProcessParameterIO not tapped. Once that is solved I won't have anything more to add.

**Comment by jdeolive** [comment]
Cool, thanks @aaime. I'll add the header to the test file and update findAll along with a test. And also to clarify whenever modifying an existing file the header needs to be updated to the current year? Is that correct?

**Comment by aaime** [comment]
@jdeolive correct, as per our commit guidelines: https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md#commit-guidelines
Doing it by hand is ok for smaller commits, for larger ones if you check geoserver-devel archives some scripts have been circulated to automate the update.

**Comment by jdeolive** [comment]
Cool, thanks. I'll refresh myself on the commit guidelines.

**Pull request merged**